pkg/component/kubernetes/apiserver/authorization.go

```diff
@@ -0,0 +1,135 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package apiserver
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	apiserverv1beta1 "k8s.io/apiserver/pkg/apis/apiserver/v1beta1"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kubernetesutils "github.com/gardener/gardener/pkg/utils/kubernetes"
+	versionutils "github.com/gardener/gardener/pkg/utils/version"
+)
+
+const (
+	volumeNameStructuredAuthorizationConfig             = "authorization-config"
+	volumeNameStructuredAuthorizationWebhookKubeconfigs = "authorization-kubeconfigs"
+
+	volumeMountPathStructuredAuthorizationConfig             = "/etc/kubernetes/structured/authorization"
+	volumeMountPathStructuredAuthorizationWebhookKubeconfigs = "/etc/kubernetes/structured/authorization-kubeconfigs" // #nosec G101 -- No credential.
+
+	configMapAuthorizationConfigDataKey = "config.yaml"
+)
+
+func (k *kubeAPIServer) reconcileConfigMapAuthorizationConfig(ctx context.Context, configMap *corev1.ConfigMap) error {
+	if value, ok := k.values.FeatureGates["StructuredAuthorizationConfiguration"]; (ok && !value) ||
+		versionutils.ConstraintK8sLess130.Check(k.values.Version) {
+		return nil
+	}
+
+	authorizationConfiguration := &apiserverv1beta1.AuthorizationConfiguration{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: apiserverv1beta1.ConfigSchemeGroupVersion.String(),
+			Kind:       "AuthorizationConfiguration",
+		},
+		Authorizers: []apiserverv1beta1.AuthorizerConfiguration{{Type: "RBAC", Name: "rbac"}},
```

Comment:
I think it would be beneficial to document that RBAC and Node are still configured behind the scenes.